staging/src/k8s.io/code-generator/cmd/client-gen/generators/fake/generator_fake_for_type.go

```diff
@@ -449,7 +434,7 @@ var updateSubresourceTemplate = `
 func (c *Fake$.type|publicPlural$) Update($.type|private$Name string, $.inputType|private$ *$.inputType|raw$) (result *$.resultType|raw$, err error) {
 	obj, err := c.Fake.
 		$if .namespaced$Invokes($.NewUpdateSubresourceAction|raw$($.type|allLowercasePlural$Resource, "$.subresourcePath$", c.ns, $.inputType|private$), &$.inputType|raw${})
-		$else$Invokes($.NewRootUpdateAction|raw$($.type|allLowercasePlural$Resource, $.type|private$), &$.type|raw${})$end$
+		$else$Invokes($.NewRootUpdateSubresourceAction|raw$($.type|allLowercasePlural$Resource, "$.subresourcePath$", $.inputType|private$), &$.resultType|raw${})$end$
```

Comment:
could you explain why the namespaced one (the line above) and the non-namespaced one (this line) differ at the end (`&$.inputType|raw${})` vs `&$.resultType|raw${})`)?

Reply:
Because the result type is always the same as the input type currently, so although it's not technically correct it won't break until you have a scenario where the update call returns a different object. 

Ex: 
``` //+genclient:method=UpdateScale,verb=update,subresource=scale,input=k8s.io/kubernetes/pkg/apis/autoscaling.Scale,result=k8s.io/kubernetes/pkg/apis/autoscaling.Scale```